### DISCUSSION TITLE
flash.sh only giving option to flash atmega16u2 MCU, and nothing else

### DISCUSSION THREAD

**Discussion by n3wkular** [shanteacontrols/OpenDeck#156] [General]
Running Ubuntu 20.04, Arduino IDE 1.8.19, and Arduino AVR Boards 1.8.5.

When I execute flash.sh, it only gives a single option for the MCU to flash:
```~/OpenDeck$ ./scripts/flash.sh 
Please type the serial port on which ArduinoISP is connected, without /dev/ part.
Port name usually has the following pattern, where X is number representing the port:
    * Linux: ttyACMX (or ttyUSBX)
    * macOS: cu.usbmodemXXX
If unsure, open Arduino IDE and check where your Arduino as an programmer is connected to under Tools/Port.
Port: ttyACM0
Please select AVR MCU you want to flash and then press enter:
     1	atmega16u2
Board number: 
```
However, I've already flashed that MCU successfully and now I need to flash the atmega2560 MCU. The port is correct. And if I select the 2560 .hex, avrdude obviously detects a device signature mismatch because the script is passing the 16u2 device to the command.

Looking at #152 , I don't really understand how to do this manually with the info there and at https://github.com/shanteacontrols/OpenDeck/blob/master/config/mcu/atmega2560.yml#L5. 

Suggestions on how to proceed? I have 4 other Mega2560s working just fine, but they were flashed over a year ago. Thanks!

**Comment by n3wkular** [comment]
Only scratching the surface here, but this was a fairly large commit: https://github.com/shanteacontrols/OpenDeck/commit/9dc89808568d2cf17fe2f067bd758ba183b19a7d. It seems the script is trying to enumerate the children of https://github.com/shanteacontrols/OpenDeck/tree/master/src/board/arch/avr/atmel/variants/avr8/, but there's not much there.

Like I said: I didn't dig very deep and you know far more than I do about how this all works. Thanks in advance for any help!

**Comment by paradajz** [comment]
Fixed. https://github.com/shanteacontrols/OpenDeck/commit/1836eea8dc3a7196b6e35a9665437794f83a5f46

The issue was that most of the generic board/ structure was [moved to core submodule](https://github.com/shanteacontrols/OpenDeck/commit/813a6006adec27c4fad24716f19d5f3a0cc682be), so some directories aren't present anymore where they used to be.

**Reply by n3wkular** [reply]
Thanks for the quick response! (Why are you awake at this hour?! :) )

So close, but still getting this:
```
Please select AVR MCU you want to flash and then press enter:
     1	atmega2560
     2	at90usb1286
     3	atmega16u2
MCU number: 1

Please wait...

./scripts/flash.sh: line 41: config/mcu/atmega2560.yml.yml: No such file or directory
./scripts/flash.sh: line 42: config/mcu/atmega2560.yml.yml: No such file or directory
./scripts/flash.sh: line 43: config/mcu/atmega2560.yml.yml: No such file or directory
./scripts/flash.sh: line 44: config/mcu/atmega2560.yml.yml: No such file or directory
./scripts/flash.sh: line 45: config/mcu/atmega2560.yml.yml: No such file or directory
Type the path (location) of the binary you want to flash.
If you want to flash official binary from OpenDeck repository, go to the following link and
download appropriate .hex file: https://github.com/shanteacontrols/OpenDeck/releases
.hex file can be found in release assets.

Tip: you can also drag .hex file to the terminal - path to the binary will be printed.
Path: '/home/rocket/OpenDeck/mega2560.hex' 
Connect programmer to programming header on the board and then press enter.

avrdude: AVR Part "atmega2560.yml" not found.
```

**Reply by paradajz** [reply]
Ah. Fixed again.

**Reply by n3wkular** [reply]
Excellent. I reflashed both MCUs successfully. Now using the newest version of the configurator for the first time ... very well done!

Cheers again for all the work and the help.

**Reply by paradajz** [reply]
No problem. Consider posting images of your controllers to Show and Tell Discussions section. :)
